core/src/main/java/org/bitcoinj/core/AddressV1Message.java

```diff
@@ -53,6 +53,17 @@ public void addAddress(PeerAddress address) {
         addresses.add(address);
     }
 
+    @Override
+    public int messageSize() {
+        if (addresses == null)
+            return 0;
```

Comment:
Can we guarantee that `addresses` is never null, but is an empty list when there are zero addresses? This could be a separate PR (either before or after this one)

Reply:
I stumbled across this as well. It may be a (pre-existing) bug.